app/src/components/ArtworkInfo.tsx

```diff
@@ -41,102 +41,89 @@ interface ArtworkInfoState {
+    title: '',
+    artistId: 0,
+    description: '',
+    edition: '',
+    artifactCreationDate: '',
+    medium: '',
+    width: '',
+    height: '',
+    imageIpfsHash: '',
+    documents: [],
+  });
+
+  const { Artists } = useContractContext();
+
+  const hashToArtist = async (hash: string): Promise<Artist> => {
     const response = await fetch(hash);
     return response.json();
   };
 
-  getArtistInfo = (): void => {
-    if (!this.state.fields.artistId && this.state.retrievedData) {
-      return;
-    }
-
-    this.props.contracts.Artists.methods.getArtist(this.state.fields.artistId)
-      .call()
-      .then((hash: string) => this.hashToArtist(hash))
-      .then((artist: Artist) => this.setState({
-        retrievedData: true,
-        artist: artist,
-      }))
-      .catch((err: any) => console.log(err));
-  };
+  React.useEffect(() => {
+    const getArtistInfo = (): void => {
+      if (!fields.artistId && retrievedData) {
+        return;
+      }
+
+      Artists.methods.getArtist(fields.artistId)
+        .call()
+        .then((hash: string) => hashToArtist(hash))
+        .then((artist: Artist) => {
+          setArtist(artist);
+          setRetrievedData(true);
+        })
+        .catch(console.log);
+    };
 
-  render (): React.ReactNode {
-    const fields = this.state.fields;
-
-    const imgSrc = fields.imageIpfsHash === ''
-      ? 'https://file.globalupload.io/HO8sN3I2nJ.png'
-      : 'https://ipfs.io/ipfs/' + fields.imageIpfsHash;
-
-    if (this.state.retrievedData) {
-      return (
-        <ArtworkCard
-          id={this.props.id}
-          img={imgSrc}
-          metaUri={this.props.artwork.metaUri}
-          fields={this.state.fields}
-          artist={this.state.artist}
-          fullscreen={this.props.fullscreen}
-        >
-          {this.props.children}
-        </ArtworkCard>
-      );
-    } else {
-      return (
-        <ArtworkCard img={imgSrc} fullscreen={this.props.fullscreen}>
-          {this.props.children}
-        </ArtworkCard>
-      );
-    }
-  }
-}
+    const setInfoFromJson = async (): Promise<void> => {
+      const metaUri = artwork.metaUri;
+
+      const response = await fetch(metaUri);
+      const infoJson = await response.json();
+      console.log(infoJson);
+      setFields(infoJson);
+
+      getArtistInfo();
+    };
+    setInfoFromJson();
+  }, [Artists, artwork.metaUri, fields, retrievedData]);
+
+  const imgSrc = fields.imageIpfsHash === ''
+    ? 'https://file.globalupload.io/HO8sN3I2nJ.png'
+    : 'https://ipfs.io/ipfs/' + fields.imageIpfsHash;
+
+  if (retrievedData) {
+    return (
+      <ArtworkCard
+        id={id}
+        img={imgSrc}
+        metaUri={artwork.metaUri}
+        fields={fields}
+        artist={artist}
+        fullscreen={fullscreen}
+      >
+        {children}
+      </ArtworkCard>
+    );
+  } else {
+    return (
+      <ArtworkCard img={imgSrc} fullscreen={fullscreen}>
+        {children}
+      </ArtworkCard>
+    );
+  };
```

Comment:
-10,000 marks

Reply:
:(

Reply:
tbf according to git blame neil wrote the -10000 marks bit a month ago we so it's an old bug